lib/AST/ASTWalker.cpp

```diff
@@ -1357,6 +1363,11 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
   }
 
   Expr *visitMacroExpansionExpr(MacroExpansionExpr *E) {
+    for (TypeRepr *genArg : E->getGenericArgs()) {
+      if (doIt(genArg))
+        return nullptr;
+    }
+
```

Comment:
Should we only walk the generic arguments if `shouldWalkArguments` is `true`? Perhaps it may also be worth factoring out the logic to walk the argument list and generic args such that it can be used by both MacroExpansionExpr and MacroExpansionDecl

Reply:
You mean the arguments portion of `shouldWalkMacroArgumentsAndExpansion`? I couldn't decide, maybe?

Reply:
Yeah, I was just thinking it may seem unexpected to walk the generic arguments if you set it to `MacroWalking::Expansion`